### ISSUE TITLE
Reconsider the softmax distribution

### ISSUE THREAD

**Issue by cheind** [cheind/autoregressive#24]
PixelCNN++ argues that neighboring intensity usually correlate is not captured by the softmax distribution. Instead, they propose a mixture model consisting of the logitic distribution (like normal but with heavier tails).
https://arxiv.org/pdf/1701.05517.pdf

**Comment by cheind** [comment]
also add bits-per-dim metric that measures how many bits are required to encode pixel intensities.

**Comment by cheind** [comment]
See also for how to optimize based on cdf
```https://github.com/Rayhane-mamah/Tacotron-2/issues/155```
